ci_scripts/utils/definitions.py

```diff
@@ -1,14 +1,93 @@
+"""
+Project definitions
+"""
 import os
+import toml
+import sys
 
-PROJECT_ROOT = os.path.realpath(
-    os.path.join(os.path.dirname(os.path.realpath(__file__)), '..',
-                 '..'))
+CONFIG_SECTION = 'ProjectConfig'
 PROJECT_CONFIG = os.path.join(os.path.dirname(__file__), '..',
                               "pyproject.toml")
-NEWS_DIR = os.path.realpath(os.path.join(PROJECT_ROOT, 'news'))
-BETA_BRANCH = 'beta'
-MASTER_BRANCH = 'master'
-RELEASE_BRANCH_PATTERN = r"^release.*$"
-REMOTE_ALIAS = 'origin'
+PATH_TOKEN = {'DIR', 'ROOT', 'PATH'}
+PROJECT_NAME = None
+PROJECT_ROOT = None
+NEWS_DIR = None
+BETA_BRANCH = None
+MASTER_BRANCH = None
+RELEASE_BRANCH_PATTERN = None
+REMOTE_ALIAS = None
+
+LOGGER_FORMAT = None
 ENVVAR_GIT_TOKEN = 'GITHUB_TOKEN'
-LOGGER_FORMAT = '%(levelname)s: %(message)s'
+
+
+class ConfigLoader:
+    """
+    Loads configuration values from `PROJECT_CONFIG` file
+    and sets the different variables.
+
+    """
+
+    class __ConfigLoader:
+        def __init__(self) -> None:
+            self.has_loaded: bool = False

```

Comment:
Should this be an internal variable? For example:

```suggestion
            self._has_loaded: bool = False

```